Dev Tools

lcaple opened the pull request:
This has been sitting staged on my computer for months so finally posting.

I kept it simple to scss and folks can add JS and other stuff if they'd like? What do you both think?

Let me know if you have recommendations.

lcaple committed:
adding dev tools folder for optional use

lcaple committed:
updated readme to add getting started with dev tools info

tjsuw reviewed (changes requested):
This is awesome! I followed the steps and it mostly went smoothly. I think this will make the build tools in the child theme really approachable to folks.

The one hiccup I ran into is having the Sass file empty and then running `gulp` wiped out the required header info at the top of the style.css file and broke the child theme. Once I added the header stuff to the Sass file, it worked as expected.

I recommend including all that header info in the Sass file and then maybe in the README suggest folks edit the Theme Name and Text Domain fields so it is unique to their site(s). There are probably other fields they want to edit too but those are the 2 biggies. As long as they don't touch the Template field! 😁 

Also I wanted to note because we were talking about this elsewhere, I have node 12.22.12 and npm 8.12.1 running and it's 👍 with the build tools. Not sure if we want to note node/npm versions or how we would talk about it (min/max versions?) - perhaps a bigger convo.

lcaple committed:
updated for compatibility with more node versions

lcaple commented:
@tjsuw I added some additional items to the ReadMe per your feedback. 

I also adjusted the npm tools included in this to get it working on more versions of node. On my machine I was able to get it functioning on node 12, 16, and 18 (all lts). I believe I tested 14 too, which is also lts.

This just involved converting to dart-sass and cssnano. Pretty straight forward!

tjsuw commented:
All looks good to me!
